File: src/main/java/com/learn/todolist/infra/TodoRepository.java
package com.learn.todolist.infra;

import com.learn.todolist.domain.Todo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface TodoRepository extends JpaRepository<Todo, Long> {
    Optional<List<Todo>> findAllByUserId(UUID user);
}

File: src/main/kotlin/com/todolist/aplication/todo/repository/TodoRepository.kt
package com.todolist.aplication.todo.repository

import com.todolist.domain.todo.Todo
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface TodoRepository: JpaRepository<Todo, Long>{
    fun findAllByUserId(userId: UUID): List<Todo>?
    fun findByUuidIdAndDeletedAtIsNull(id: UUID): Todo?
    fun findByUserIdAndDeletedAtIsNullOrderByUpdatedAtDesc(userId: UUID): Todo?
    fun findByUserIdAndDeletedAtIsNullOrderByCreatedAtDesc(userId: UUID): Todo?
}